diff -ru busybox-1.18.4/shell/ash.c busybox-1.18.4-android/shell/ash.c
--- busybox-1.18.4/shell/ash.c	2011-03-13 10:45:40.000000000 +0900
+++ busybox-1.18.4-android/shell/ash.c	2011-05-12 10:37:16.020132392 +0900
@@ -13054,20 +13054,17 @@
 		const char *hp = lookupvar("HISTFILE");
 
 		if (hp == NULL) {
-			hp = lookupvar("HOME");
-			if (hp != NULL) {
-				char *defhp = concat_path_file(hp, ".ash_history");
-				setvar("HISTFILE", defhp, 0);
-				free(defhp);
-			}
+			setvar("HISTFILE", "/data/ash_history", 0);
 		}
 	}
 #endif
 	if (/* argv[0] && */ argv[0][0] == '-')
 		isloginsh = 1;
+	else
+		isloginsh = 1;
 	if (isloginsh) {
 		state = 1;
-		read_profile("/etc/profile");
+		read_profile("/system/etc/profile");
  state1:
 		state = 2;
 		read_profile(".profile");
